proxy/src/dns.rs

```diff
@@ -73,70 +66,92 @@ impl AsRef<str> for Name {
 }
 
 impl Config {
-    /// Note that this ignores any errors reading or parsing the resolve.conf
-    /// file, just like the `domain` crate does.
-    pub fn from_file(resolve_conf_path: &Path) -> Self {
-        let mut resolv_conf = domain::resolv::ResolvConf::new();
-        let _ = resolv_conf.parse_file(resolve_conf_path);
-        resolv_conf.finalize();
-        Config(resolv_conf)
+    /// TODO: Make this infallible, like it is in the `domain` crate.
```

Comment:
This is one of the bigger risks in this PR. The `domain` crate used a parser for /etc/resolv.conf that was infallible; trust-dns-resolver doesn't have an infallible parser. OTOH, parsing junk input is pretty dangerous and difficult to reason about. In practice we expect Kubernetes will write valid configuration files.

Reply:
To me, it seems that this function being falliable is more correct --- what would happen if the `domain` parser were to read junk input? I suspect that whatever the answer to that question is, it's probably worse than unwrapping this result & bailing with a descriptive message.